assets/js/data/cart/push-changes.ts

```diff
@@ -27,15 +23,33 @@ type CustomerData = {
 	shippingAddress: CartShippingAddress;
 };
 
+type BillingOrShippingAddress = CartBillingAddress | CartShippingAddress;
+
 /**
  * Checks if a cart response contains an email property.
  */
 const isBillingAddress = (
-	address: CartBillingAddress | CartShippingAddress
+	address: BillingOrShippingAddress
 ): address is CartBillingAddress => {
 	return 'email' in address;
 };
 
+export const trimAddress = ( address: BillingOrShippingAddress ) => {
+	const trimmedAddress = {
+		...address,
+	};
+	Object.keys( address ).forEach( ( key ) => {
+		trimmedAddress[ key as keyof BillingOrShippingAddress ] =
+			address[ key as keyof BillingOrShippingAddress ].trim();
+	} );
+
+	trimmedAddress.postcode = trimmedAddress.postcode
+		? trimmedAddress.postcode.replace( ' ', '' ).toUpperCase()
+		: '';
+
+	return trimmedAddress;
+};
+
```

Comment:
Why is trimming an address needed?

Reply:
It was done in the `pluckAddress` function, when comparing values it's good practice to `trim` to avoid false negatives